src/core/public/core_system.ts

```diff
@@ -155,27 +156,30 @@ export class CoreSystem {
 
   public async start() {
     try {
-      // ensure the rootDomElement is empty
-      this.rootDomElement.textContent = '';
-      this.rootDomElement.classList.add('coreSystemRootDomElement');
+      const injectedMetadata = this.injectedMetadata.start();
```

Comment:
Did you want to `await` these starts even though right now there is no async work in case they move to async in the future?